docs: improve clarity and formatting of emoji-key documentation

KhushiPandey8 opened the pull request:
<!--
Thanks for your interest in the project. Bugs filed and PRs submitted are appreciated!

Please make sure that you are familiar with and follow the [Code of Conduct](https://allcontributors.org/docs/en/project/code-of-conduct) for this project.

Also, please make sure you're familiar with and follow the instructions in the
[contributing guidelines](https://github.com/all-contributors/all-contributors/blob/master/CONTRIBUTING.md) (found in the CONTRIBUTING.md file).

If you're new to contributing to open source projects, you might find this free
video course helpful: http://kcd.im/pull-request

Please fill out the information below to expedite the review and (hopefully)
merge of your pull request!
-->

<!-- What changes are being made? (What feature/bug is being fixed here?)

e.g. Fixes #0

Check this [list](https://help.github.com/en/articles/closing-issues-using-keywords) of valid keywords.
 -->
**What**:
Improved clarity, formatting, and wording of the emoji-key documentation.

<!-- Why are these changes necessary? -->
**Why**:
To make contribution types easier to understand without changing any functionality.

<!-- How were these changes implemented? -->
**How**:
- Refined explanations
- Improved table structure and consistency
- Added clarification note about supported contribution types
- No new types added

<!-- Have you done all of these things?  -->
**Checklist**:
<!-- add "N/A" to the end of each line that's irrelevant to your changes -->
<!-- to check an item, place an "x" in the box like so: "- [x] Documentation" -->
- [ ] Documentation
- [ ] Ready to be merged <!-- In your opinion, is this ready to be merged as soon as it's reviewed? -->
- [ ] Added myself to contributors table. <!-- this is optional, see the contributing guidelines for instructions -->
[Bot Usage](https://allcontributors.org/docs/en/bot/installation#4-update-your-contributing-documentation)

<!-- feel free to add additional comments -->

KhushiPandey8 committed:
Update emoji-key.md

Extended This Emoji-key list

KhushiPandey8 committed:
Update emoji-key.md

Improve the readability and utility of the document

lwasser commented:
@all-contributors please add @KhushiPandey8 for docs

lwasser commented:
Similar to the previous PR that I merged, we should merge this and then pull it into our docs-upgrade branch via a cherry pick . so i'm not going to merge until i'm ready to do that so we don't lose the updates. Thank you @KhushiPandey8 for this PR!! ❤️

lwasser commented:
Oops - it looks like we need to lint the pr as well before we can merge.

KhushiPandey8 committed:
Revise Emoji Key document for clarity and examples

Updated the title and clarified the purpose of the Emoji Key document. Added examples for contribution types and improved formatting.

JimMadge committed:
chore: resolve changes with moved file

JimMadge reviewed (changes requested):
I like the changes to the table.

I think there is a little work to be done to make sure the language is consistent with the rest of the documentation.

There is also some duplication on this page when talking about natural language processing and the bot commands. I think here we can just detail the types and briefly mention you will need to use these when using the bot or CLI. What do you think @lwasser?

KhushiPandey8 committed:
Fix links and reorganize emoji key documentation

Updated the required changes

KhushiPandey8 committed:
Merge branch 'main' into patch-1

KhushiPandey8 committed:
Fixed the arising issue due to styling

JimMadge committed:
Correct admonition

JimMadge committed:
Remove out of scope text

JimMadge committed:
Fix typo

JimMadge reviewed (approved):
Thanks @KhushiPandey8 :tada:
